### ISSUE TITLE
Doesn't work with webpack.

### ISSUE THREAD

**Issue by reggi** [desandro/masonry#679]
I'm trying to use the NPM version with webpack and I get the following errors.

```
ERROR in ./~/masonry-layout/~/outlayer/outlayer.js
Module not found: Error: Cannot resolve module 'eventEmitter/EventEmitter' in /Users/thomas/Desktop/browserify-deamdify-test/node_modules/masonry-layout/node_modules/outlayer
 @ ./~/masonry-layout/~/outlayer/outlayer.js 989:2-997:24

ERROR in ./~/masonry-layout/~/outlayer/outlayer.js
Module not found: Error: Cannot resolve module 'matches-selector/matches-selector' in /Users/thomas/Desktop/browserify-deamdify-test/node_modules/masonry-layout/node_modules/outlayer
 @ ./~/masonry-layout/~/outlayer/outlayer.js 989:2-997:24

ERROR in ./~/masonry-layout/~/get-size/get-size.js
Module not found: Error: Cannot resolve module 'get-style-property/get-style-property' in /Users/thomas/Desktop/browserify-deamdify-test/node_modules/masonry-layout/node_modules/get-size
 @ ./~/masonry-layout/~/get-size/get-size.js 241:2-70

ERROR in ./~/masonry-layout/~/outlayer/item.js
Module not found: Error: Cannot resolve module 'eventEmitter/EventEmitter' in /Users/thomas/Desktop/browserify-deamdify-test/node_modules/masonry-layout/node_modules/outlayer
 @ ./~/masonry-layout/~/outlayer/item.js 507:2-512:28

ERROR in ./~/masonry-layout/~/outlayer/item.js
Module not found: Error: Cannot resolve module 'get-style-property/get-style-property' in /Users/thomas/Desktop/browserify-deamdify-test/node_modules/masonry-layout/node_modules/outlayer
 @ ./~/masonry-layout/~/outlayer/item.js 507:2-512:28
```

It looks like there's modules that aren't properly being defined.

**Comment by reggi** [comment]
I ran `npm install get-style-property --save` and got rid of those messages. Running `npm i desandro-matches-selector --save` didn't do anything.

**Comment by reggi** [comment]
Because webpack process AMD code but looks for NPM modules [it breaks here](https://github.com/metafizzy/outlayer/blob/master/outlayer.js#L987-L1008) in outlayer.
```
if ( typeof define === 'function' && define.amd ) {
  // AMD
  define( [
      'eventie/eventie',
      'doc-ready/doc-ready',
      'eventEmitter/EventEmitter',
      'get-size/get-size',
      'matches-selector/matches-selector',
      './item'
    ],
    outlayerDefinition );
} else if ( typeof exports === 'object' ) {
  // CommonJS
  module.exports = outlayerDefinition(
    require('eventie'),
    require('doc-ready'),
    require('wolfy87-eventemitter'),
    require('get-size'),
    require('desandro-matches-selector'),
    require('./item')
  );
```

**Comment by desandro** [comment]
Thanks for reporting this bug. Here's what I've found to work.

Install Masonry and [imports-loaded](https://github.com/webpack/imports-loader) with npm.

``` bash
npm install masonry-layout
npm install imports-loader
```

In your [config file](http://webpack.github.io/docs/configuration.html), `webpack.config.js`, use the [`imports` loader](https://github.com/webpack/imports-loader) to disable `define` and set `window` for `masonry-layout` modules.

``` js
module.exports = {
  module: {
    loaders: [
      {
        test: /masonry-layout/,
        loader: 'imports?define=>false&this=>window'
      }
    ]
  }
};
```

You should be able to `require('masonry-layout')` no problem.

``` js
// main.js
var Masonry = require('masonry-layout');

var msnry = new Masonry( '.grid', {
  itemSelector: '.grid-item',
  columnWidth: 200
});
```

``` bash
webpack main.js bundle.js
```

**Comment by reggi** [comment]
This is a tad better then my solution! Thanks!

```
module.exports = {
  entry: "./main.js",
  output: {
    filename: "./pack.js"
  },
  resolve:{
    modulesDirectories: ['node_modules'],
    alias:{
      "matches-selector/matches-selector": "desandro-matches-selector",
      "eventEmitter/EventEmitter": "wolfy87-eventemitter",
      "get-style-property/get-style-property": "desandro-get-style-property"
    }
  }
};
```

**Comment by davidgilbertson** [comment]
@desandro Brilliant, thanks for that!

**Comment by TCMiranda** [comment]
The solution should not require the user of Webpack to define a custom rule to build Masonry...
Why can't those packages be bundled together?

**Comment by desandro** [comment]
@TCMiranda Masonry and its dependencies are designed for interoperability with multiple package loaders. The underlying issue is with how Webpack uses the first export it sees in the source code. Take a look at webpack/webpack#883, and please add your support there.

**Comment by desandro** [comment]
Masonry v4.0 and fizzy-ui-utils v2.0.1 has a fix for this issue, using the same package names for AMD and CommonJS dependencies. The `config` hack code is no longer necessary. You should be able to `npm install masonry-layout` and `require('masonry')` no problem.

Closing as fixed :tada:

**Comment by parrott-kevin** [comment]
@desandro any idea when the fizzy-ui-utils fix will be published (I'm assuming it's related to PR #1 as making that fix manually seems to fix the issue)?

**Comment by desandro** [comment]
fizzy-ui-utils v2.0.1 is already published. Try `npm update`
